### ISSUE TITLE
Feature request: add `\tuplet` keyword handling in abjad

### ISSUE THREAD

**Issue by rkubosz** [Abjad/abjad#918]
Hello!
It seems, that parser understands tuplets only initialized with `\times` keyword. Since Lilypond v2.18.2 it is possible to write tuplets with `\tuplet`. [[source](http://lilypond.org/doc/v2.18/Documentation/notation/writing-rhythms#tuplets)].
Is there any chance that abjad will handle `\tuplet` keyword both in parser and in lilypond code output?
I might try to add this feature, but I would need some help with it since I barely know where to start editing abjad source code.

**Comment by trevorbaca** [comment]
Hi @rkubosz!

What type of ``\tuplet`` support are you thinking of?

Abjad tuplets initialize with a tuplet multiplier, as you were pointing out:
```
tuplet = abjad.Tuplet((2, 3), "c'4 d' e'")
abjad.f(tuplet)
\times 2/3 {
    c'4
    d'4
    e'4
}
```
This way of working with tuplets is equivalent to thinking of tuplets as carrying a multiplier, like 2/3, that time-scales the music in a tuplet.

As per endless years of discussion on the LilyPond list (!), this multiplier-oriented way of thinking about tuplets runs counter to the "two-in-the-time-of-three" way that almost everybody first learns to play these rhythms. We might call this (much more usual) way of working with tuplets "ratio-oriented"; this helps us remember that when we're thinking two-in-the-time-of-three or five-in-the-time-of-seven we're thinking ratios like 2:3 or 5:7 (as opposed to multipliers like 2/3 or 5/7).

So how to have it both ways?

First note that Abjad tuplets can initialize with `abjad.Multiplier` objects directly:
```
tuplet = abjad.Tuplet(abjad.Multiplier(2, 3), "c'4 d' e'"))
abjad.f(tuplet)
\times 2/3 {
    c'4
    d'4
    e'4
}
```

And then note that the relationship between tuplet ratios and tuplet multipliers is straightforward because each is the reciprocal of the other:
```
multiplier = abjad.Multiplier((2, 3))
multiplier.reciprocal
Multiplier(3, 2)
```

So one way of working with Abjad tuplets in a ratio-oriented way is to carry around `abjad.Multiplier` objects in your score code and take the reciprocal at tuplet initialization-time:
```
ratio = abjad.Multiplier((3, 2)) # thinking two-in-the-time-of-three
...
# ... many lines of composing go here ...
...
tuplet = abjad.Tuplet(ratio.reciprocal, "c'4 d' e'')
abjad.f(tuplet)
\times 2/3 {
    c'4
    d'4
    e'4
}
```
Maybe this gets you what you're looking for? You can work with objects that 'look like' 3:2 and get the output you'd expect.

Let me know what else you were thinking. Will keep issue open for a while.

**Comment by rkubosz** [comment]
Hi @trevorbaca!
I thought about supporting `\tuplet` in parser/lexer. I did my research and discovered, that there is a huge Josiah's branch (PR #538)  which adds support for Lilypond 2.19 features, including the `\tuplet` keyword in parser. I thought that I might try to merge it, but it needs a lot of work and knowledge about the project, so I gave up.

Eventually, I decided to write a python script that converts `\tuplet 3/2` to `\times 2/3` and for purpose of processing music in abjad it is a sufficient solution for my needs.

Thanks for your kind reply! I think that this issue can be closed, since there is a solution for this but not merged yet :-)
